### ISSUE TITLE
Pollack inconsistency when module name is incorrect

### ISSUE THREAD

**Issue by catalin-hritcu** [FStarLang/FStar#560]
As discovered with @nikswamy  and @msprotz in https://github.com/FStarLang/FStar/issues/399#issuecomment-228450512 when the name of the module doesn't match the name of the file F* is Pollack inconsistent: it says it verified the file, but in fact it really did nothing. It does print a warning:
```
Warning: the module declaration "module Bug" found in file bug399.fst does not match its filename. Dependencies will be incorrect.
```
but this only talks about dependencies and does not clearly reflect the gravity of the problem. Discussed with @msprotz about turning this warning into an error at least. This would require fixing file/module names everywhere.

**Comment by nikswamy** [comment]
The proposed fix is in auto-deps mode: when the warning is issued, so long as the user has not already specified a `--verify_module` option on the command line herself, we will add `--verify_module B` where `B` is the name of module for which the warning is raised.

**Comment by nikswamy** [comment]
This proposed fix will not require changing existing files/module names.

**Comment by s-zanella** [comment]
It would be great if you could also fix #415 while you are at it.

**Comment by msprotz** [comment]
Fixed via f81093399b85245a4cdc01d0e14c8dbd8e76e1cb
